### PR TITLE
Add Sway port

### PR DESCRIPTION
**Pull request by no92**
Blocked on managarm/managarm#449, managarm/managarm#419 and managarm/mlibc#367.

swaybar is known not to work yet.

### CODE REVIEW

**Comment by Dennisbonke** [comment]
I intend to review this more in depth later today, but why is Xwayland not working?

File: patches/seatd/0001-Add-managarm-specific-changes.patch
Comment (avdgrinten) [inline patches/seatd/0001-Add-managarm-specific-changes.patch:159]: Why do we patch this out?

File: patches/wlroots/0001-Add-managarm-specific-changes.patch
Comment (avdgrinten) [inline patches/wlroots/0001-Add-managarm-specific-changes.patch:58]: This doesn't work outside of our standard rootfs. We need to fix this in a more general way.

File: bootstrap.d/gui-apps.yml
Comment (Dennisbonke) [inline bootstrap.d/gui-apps.yml:14]: We shouldn't hardcode any arch here, use `triple: "@OPTION:arch-triple@"` instead.

File: bootstrap.d/gui-libs.yml
Comment (Dennisbonke) [inline bootstrap.d/gui-libs.yml:15]: We shouldn't hardcode any arch here, use `triple: "@OPTION:arch-triple@"` instead.

File: scripts/limine.cfg
Comment (Dennisbonke) [inline scripts/limine.cfg:21]: Is this needed for this PR or a remnant of our efforts to get pure X working? In the last case, we can remove it.

File: bootstrap.d/gui-apps.yml
Comment (Dennisbonke) [inline bootstrap.d/gui-apps.yml:7]: Why are we building from a commit? There are releases for swaybg that we can use right?

File: bootstrap.d/gui-apps.yml
Comment (Dennisbonke) [inline bootstrap.d/gui-apps.yml:3]: Missing metadata information, also lacking `architecture: '@OPTION:arch@'`. While not all ports have this, we should add it when possible and I'd like new ports to do it by default. This is also the moment where you can decide if you like to be the maintainer of this port.

File: patches/wlroots/0001-Add-managarm-specific-changes.patch
Comment (ArsenArsen) [reply patches/wlroots/0001-Add-managarm-specific-changes.patch:58]: why is this even needed? isn't this discovered via .pcs

File: bootstrap.d/gui-apps.yml
Comment (Dennisbonke) [inline bootstrap.d/gui-apps.yml:26]: Why are the man pages disabled here?

File: bootstrap.d/gui-apps.yml
Comment (Dennisbonke) [inline bootstrap.d/gui-apps.yml:27]: Why is gdk-pixbuf disabled? We have that ported so we can use it.

File: bootstrap.d/gui-apps.yml
Comment (Dennisbonke) [inline bootstrap.d/gui-apps.yml:15]: Missing direct dependency on `mlibc`.

File: bootstrap.d/gui-libs.yml
Comment (Dennisbonke) [inline bootstrap.d/gui-libs.yml:3]: Missing metadata information, also lacking `architecture: '@OPTION:arch@'`. While not all ports have this, we should add it when possible and I'd like new ports to do it by default. This is also the moment where you can decide if you like to be the maintainer of this port.

File: bootstrap.d/gui-libs.yml
Comment (Dennisbonke) [inline bootstrap.d/gui-libs.yml:18]: Missing direct dependency on `mlibc`.

File: bootstrap.d/gui-wm.yml
Comment (Dennisbonke) [inline bootstrap.d/gui-wm.yml:3]: Missing metadata information, also lacking `architecture: '@OPTION:arch@'`. While not all ports have this, we should add it when possible and I'd like new ports to do it by default. This is also the moment where you can decide if you like to be the maintainer of this port.

File: bootstrap.d/gui-wm.yml
Comment (Dennisbonke) [inline bootstrap.d/gui-wm.yml:18]: Missing direct dependency on `mlibc`.

File: bootstrap.d/gui-wm.yml
Comment (Dennisbonke) [reply bootstrap.d/gui-wm.yml:40]: Bash completions aren't ported yet so I'm unsure if they would do anything. I plan on rectifying that soon tho.

File: bootstrap.d/gui-wm.yml
Comment (Dennisbonke) [inline bootstrap.d/gui-wm.yml:39]: Why is gdk-pixbuf disabled if we have it?

File: bootstrap.d/sys-auth.yml
Comment (Dennisbonke) [inline bootstrap.d/sys-auth.yml:5]: Missing metadata information. While not all ports have this, we should add it when possible and I'd like new ports to do it by default. This is also the moment where you can decide if you like to be the maintainer of this port.

File: bootstrap.d/x11-libs.yml
Comment (Dennisbonke) [inline bootstrap.d/x11-libs.yml:221]: Why is this change needed? Also, if this is actually something we need, don't forget to bump the revision, otherwise the ci won't pick it up.

File: extrafiles/sway-config
Comment (Dennisbonke) [inline extrafiles/sway-config:68]: We should either set `$menu` or comment this line out for now.

File: patches/seatd/0001-Add-managarm-specific-changes.patch
Comment (Dennisbonke) [inline patches/seatd/0001-Add-managarm-specific-changes.patch:179]: I presume this was done to remove warnings, but assuming the `#else` is part of the patched `#if defined(__linux__) || defined(__managarm__)` then this shouldn't be needed I believe?

**Review by Dennisbonke — CHANGES_REQUESTED** [review]
Several changes, mostly to please the package maintainer inside me. Also going to block this on Xwayland not working, as it _should_ just work:tm:.

File: patches/wlroots/0001-Add-managarm-specific-changes.patch
Comment (no92) [reply patches/wlroots/0001-Add-managarm-specific-changes.patch:58]: wlroots states that xwayland just doesn't ship `.pc` files, so we need this. Using `sed` to replace a variable seems like the reasonable solution, though.

File: bootstrap.d/gui-apps.yml
Comment (no92) [reply bootstrap.d/gui-apps.yml:39]: scdoc is required, which we don't have.

File: bootstrap.d/gui-apps.yml
Comment (no92) [reply bootstrap.d/gui-apps.yml:40]: post-install hooks are not quite working, so that's off the table right now.

File: patches/wlroots/0001-Add-managarm-specific-changes.patch
Comment (ArsenArsen) [inline patches/wlroots/0001-Add-managarm-specific-changes.patch:58]:
```
[i] ~/managarm/bootstrap-managarm$ pkg-config --list-all | grep xwayland
xwayland                       Xwayland - X Server for Wayland
[i] ~/managarm/bootstrap-managarm$
```

**Review by ArsenArsen — CHANGES_REQUESTED** [review]

File: bootstrap.d/gui-apps.yml
Comment (Dennisbonke) [reply bootstrap.d/gui-apps.yml:40]: These have been fixed now, can we enable this?

File: bootstrap.d/gui-wm.yml
Comment (Dennisbonke) [inline bootstrap.d/gui-wm.yml:5]: Summary should start with a capital.

File: bootstrap.d/gui-wm.yml
Comment (Dennisbonke) [reply bootstrap.d/gui-wm.yml:48]: See my other gdk-pixbuf comment.

**Review by Dennisbonke — COMMENTED** [review]
Looks good, few small things here and there.

**Review by Dennisbonke — CHANGES_REQUESTED** [review]
Patch for sway and wlroots does not apply, blocked on 1.8 release and wlroots is missing a commit.

**Comment by no92** [comment]
Updated to current sway master with a pinned commit as there is no 1.8 release yet. Comments were addressed, as xwayland now uses pkg-config.

**Comment by ArsenArsen** [comment]
Why?

**Comment by no92** [comment]
Sway 1.7 has issues that require more extensive patching and different versions of dependencies, but no timeline for a 1.8 release is given. Pinning a commit until 1.8 is released seems like the most reasonable choice here, IMO.

**Comment by ArsenArsen** [comment]
which ones? also, I don't see a commit hash in the source, did you forget to pin it? also, this arguably isn't 1.7, nor does it include some of the 1.7 commits (they were branched off)

**Comment by ArsenArsen** [comment]
might be more apt to call this `1.7snapshot_20220930` or so; I'd have to check what xbps-create accepts (this is not 1.7 nor 1.8)

File: bootstrap.d/gui-apps.yml
Comment (Dennisbonke) [inline bootstrap.d/gui-apps.yml:15]: Style issue, we use single quotes

File: bootstrap.d/gui-libs.yml
Comment (Dennisbonke) [inline bootstrap.d/gui-libs.yml:15]: Style issue, we use single quotes

File: bootstrap.d/gui-wm.yml
Comment (Dennisbonke) [inline bootstrap.d/gui-wm.yml:14]: Style nitpick, missing single quotes

File: bootstrap.d/gui-wm.yml
Comment (Dennisbonke) [inline bootstrap.d/gui-wm.yml:12]: Style nitpick, missing single quotes

File: bootstrap.d/gui-libs.yml
Comment (Dennisbonke) [inline bootstrap.d/gui-libs.yml:12]: Style nitpick, missing single quotes

File: bootstrap.d/gui-libs.yml
Comment (Dennisbonke) [inline bootstrap.d/gui-libs.yml:14]: Style nitpick, missing single quotes

File: bootstrap.d/gui-libs.yml
Comment (Dennisbonke) [inline bootstrap.d/gui-libs.yml:8]: Style nitpick, missing single quotes

File: bootstrap.d/gui-apps.yml
Comment (Dennisbonke) [inline bootstrap.d/gui-apps.yml:8]: Style nitpick, missing single quotes

File: bootstrap.d/gui-apps.yml
Comment (Dennisbonke) [inline bootstrap.d/gui-apps.yml:12]: Style nitpick, missing single quotes

File: bootstrap.d/gui-wm.yml
Comment (Dennisbonke) [inline bootstrap.d/gui-wm.yml:8]: Style nitpick, missing single quotes

File: extrafiles/sway-config
Comment (Dennisbonke) [reply extrafiles/sway-config:38]: Still not fixed!

File: bootstrap.d/gui-wm.yml
Comment (Dennisbonke) [inline bootstrap.d/gui-wm.yml:36]: This should be `xwayland` now.

File: bootstrap.d/gui-libs.yml
Comment (Dennisbonke) [inline bootstrap.d/gui-libs.yml:35]: This should be `xwayland` now.

**Review by Dennisbonke — CHANGES_REQUESTED** [review]
Almost there! Also the `wlroots` patch seems broken.

**Comment by ArsenArsen** [comment]
btw: sway 1.8 just came out

**Comment by no92** [comment]
Updated to sway `1.8.1`.

File: bootstrap.d/gui-libs.yml
Comment (Dennisbonke) [inline bootstrap.d/gui-libs.yml:34]: Double dependency here.

File: bootstrap.d/x11-libs.yml
Comment (Dennisbonke) [inline bootstrap.d/x11-libs.yml:1851]: You forgot to remove the revision here.

**Review by Dennisbonke — CHANGES_REQUESTED** [review]
Two comments, LGTM outside that

**Review by Dennisbonke — APPROVED** [review]
LGTM

**Pull request merged**
